osu.Game/Graphics/Containers/GrowToFitContainer.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Graphics.Containers;
+
+namespace osu.Game.Graphics.Containers
+{
+    /// <summary>
+    /// A container that grows in size to fit its child and retains its size when its child shrinks
```

Comment:
Why is this needed?

Reply:
Previously (before the commit in which i put each TextSprite inside a GrowToFitContainer) the structure of SongProgressInfo was something like this:
SongProgressInfo:
    Container
        TextSprite
    Container
        TextSprite
    Container
        TextSprite
And each container has AutoSizeAxes = Axes.Both, so its size depended on the size of the TextSprite inside it, on the other hand each TextSprite is anchored to the center of the container and its origin is set to center, this means that the position of TextSprite depends on the size of the Container and the size of the Container depends on the size of the TextSprite, all of this would have been fine if the size of the TextSprite was fixed but it is not, as it depends on the text inside it which changes, causing indirectly the position of the TextSprite to change, in order to fix this it is necessary to fix the size of TextSprite which is a bad idea, so instead of doing that i put the TextSprites inside a container (GrowToFitContainer) that does not change its size every second and made it grow to fit any TextSprite size

Reply:
You can see the problem if you remove the `GrowToFitContainer` and put the the `TextSprite`s directly inside the `Container` then go to skin editor and rotate the progress bar 45 degrees, i will upload a video showing this problem.

Reply:
https://user-images.githubusercontent.com/101936124/183753341-fe309639-fc13-43f7-ab3f-a30d74bdd361.mp4

Reply:
Here is a closer look: `TextSprite` is yellow, `Container` is red, `SongProgressBarInfo` is violet

https://user-images.githubusercontent.com/101936124/183762862-9e63a55f-b9c8-4914-8833-0f1a5e8ea7c4.mp4

t

Reply:
Notice how the size of `TextSprite` depends on the text inside it, which is affecting its position.

Reply:
Well, on a quick look I unfortunately can't see a way to get rid of this `GrowToFitContainer`. But it shouldn't be required to manually instantiate one to have `UprightUnscaledContainer` work correctly. `UprightUnscaledContainer` itself should ensure that actual content is always nested in a `GrowToFitContainer`, and `GrowToFitContainer` should be a private inner class inaccessible by anything else.

There are [examples in the repository](https://github.com/ppy/osu/blob/master/osu.Game.Rulesets.Catch/UI/CatchPlayfieldAdjustmentContainer.cs) of how you'd do this - you'd generally have the `GrowToFitContainer` as the `InternalChild` of the `UprightUnscaledContainer`, and override `Content` so that it points at the `GrowToFitContainer`, ensuring that any drawable that is `.Add()`ed to the `UprightUnscaledContainer` is actually internally added to the `GrowToFitContainer`.